wafer/talks/forms.py

```diff
@@ -12,26 +12,37 @@
 
 class TalkForm(forms.ModelForm):
     def __init__(self, *args, **kwargs):
+        self.user = kwargs.pop('user')
         super(TalkForm, self).__init__(*args, **kwargs)
         self.helper = FormHelper(self)
-        submit_button = Submit('submit', _('Submit'))
+
+        actions = (Submit('submit', _('Submit')),)
+
+        if not self.user.has_perm('talks.edit_private_notes'):
+            self.fields.pop('private_notes')
```

Comment:
I think if you pop the field first, you won't need to entirely replace the layout, e.g.
```
if not self.user.has_perm('talks.edit_private_notes'):
    self.fields.pop('private_notes')
self.helper = FormHelper(self)
actions = (Submit('submit', _('Submit')),)
... possibly add delete action ...
self.helper.layout.append(FormActions(*actions))
```